engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/ValidatableParametersMetaData.java

```diff
@@ -18,14 +25,41 @@
  */
 public class ValidatableParametersMetaData implements Validatable {
 
+	private final List<ParameterMetaData> parameterMetaData;
 	private final Iterable<Cascadable> cascadables;
 
-	public ValidatableParametersMetaData(Iterable<? extends Cascadable> cascadables) {
-		this.cascadables = CollectionHelper.<Cascadable>newHashSet( cascadables );
+	public ValidatableParametersMetaData(List<ParameterMetaData> parameterMetaData) {
+		this.parameterMetaData = Collections.unmodifiableList( parameterMetaData );
+		this.cascadables = Collections.unmodifiableList( parameterMetaData.stream()
+			.filter( p -> p.isCascading() )
+			.collect( Collectors.toList() ) );
```

Comment:
I don't think it makes any effective difference at this point. I kinda liked that it keeps the order of the parameters when iterating through the cascaded ones.